app/server/lib/Authorizer.ts

```diff
@@ -203,6 +203,15 @@ export async function addRequestUser(
       if (!user) {
         return res.status(401).send('Bad request: invalid API key');
       }
+      if (user.type === "service") {
+        const serviceAccount = await dbManager.getServiceAccountWithOwner(user.loginEmail as string);
+        if (serviceAccount?.owner.disabledAt) {
+          return res.status(403).send('Unauthorized');
+        }
+        if (!serviceAccount!.isAlive()) {
```

Comment:
There should be more consistency here.

Line 208, we are not sure whether `serviceAccount` is null-ish:
```
        if (serviceAccount?.owner.disabledAt) {
```

But here, we assert that:
```
        if (!serviceAccount!.isAlive()) {
```

If we're sure that it exists, probably we should just reflect that instead line 207 by adding a `!` at the end of the line:
```
        const serviceAccount = await dbManager.getServiceAccountWithOwner(user.loginEmail as string)!;
```